### PR TITLE
updated date field docs

### PR DESCRIPTION
**Pull request by ncphillips**
- remove gatsby from example
- switch from options list ot table
- add dateFormat and timeFormat to options table

### CODE REVIEW

File: content/docs/fields/date.md [hunk 1]
```diff
@@ -13,47 +13,55 @@ The `date` field represents a date and time picker. It can be used for data valu
...
+| `timeFormat`  | Defines the format for the time. It accepts any [Moment.js time format](https://momentjs.com/docs/#/displaying/format/) (not in localized format). If true the time will be displayed using the defaults for the current locale. If false the timepicker is disabled and the component can be used as datepicker. See [react-datetime docs](https://github.com/YouCanBookMe/react-datetime) for more details._(Optional)_ |
 
-- `name`: The path to some value in the data being edited.
-- `component`: The name of the React component that should be used to edit this field. Available field component types are [defined here](/docs/fields)
-- `label`: A human readable label for the field. This label displays in the sidebar and is optional. If no label is provided, the sidebar will default to the name.
-- `description`: An optional description that expands on the purpose of the field or prompts a specific action.
+> ### FieldConfig
```
Comment (logan-anderson) [inline content/docs/fields/date.md:40]: Thanks for adding this! This confused me at first not knowing that FieldConfig was.

File: content/docs/fields/date.md [hunk 2]
```diff
@@ -13,47 +13,55 @@ The `date` field represents a date and time picker. It can be used for data valu
...
+      label: 'Date',
+      name: 'created_at',
+      component: 'date',
+      dateFormat: 'MMMM DD YYYY',
+      timeFormat: false,
+    },
+    // ...
+  ],
+}
+```
```
Comment (logan-anderson) [inline content/docs/fields/date.md:67]: Would it be worth it also linking to a real-world example like [BlogPostCreator]( https://github.com/tinacms/tinacms.org/blob/e217a45a2408ff6fd713939885f886a342953a40/tinacms/BlogPostCreator.ts#L19)  or [blog slug page](https://github.com/tinacms/tinacms.org/blob/e217a45a2408ff6fd713939885f886a342953a40/pages/blog/%5Bslug%5D.tsx#L165) (tina cms site).  That way folks could see a working example. or would this just cause problems for maintenance?

Comment (ncphillips) [reply content/docs/fields/date.md:67]: Hmm 🤔 These docs are meant to be purely Tina (i.e. no reference to frameworks or backends) but I definitely see the value in that. 

What do the other@tinacms/maintainers think?

Comment (ncphillips) [reply content/docs/fields/date.md:67]: Shit my mention didn't work.

@tinacms/maintainers

Comment (kendallstrautman) [reply content/docs/fields/date.md:67]: Yeah I'd rather just have the examples in this doc. If there are other use cases that should be noted, let's add them!

**Pull request merged**